sklearn/model_selection/_split.py

```diff
@@ -1823,16 +1823,22 @@ def _validate_shuffle_split(n_samples, test_size, train_size):
     size of the data (n_samples)
     """
     if (test_size is not None and
-            np.asarray(test_size).dtype.kind == 'i' and
-            test_size >= n_samples):
-        raise ValueError('test_size=%d should be smaller than the number of '
-                         'samples %d' % (test_size, n_samples))
+            (np.asarray(test_size).dtype.kind == 'i' and
+                (test_size >= n_samples or test_size <= 0)) or
+            (np.asarray(test_size).dtype.kind == 'f' and
+                (test_size <= 0 or test_size >= 1))):
+        raise ValueError('test_size=%d should be either positive and smaller '
+                         'than the number of samples %d or a float in the '
+                         '[0,1] range' % (test_size, n_samples))
```

Comment:
These bounds should be exclusive, i.e. `(0, 1)`

Reply:
@jnothman Why? What is the use case for either 0 or 1? It would just mean you take the whole input and return it. Isn't it so?

Reply:
`x in [0,1]` means `0 <= x <= 1` conventionally.

`x in (0,1)` means `0 < x < 1` conventionally.

We want the latter.

See https://stackoverflow.com/questions/4396290/what-does-this-square-bracket-and-parenthesis-bracket-notation-mean-first1-last and https://en.wikipedia.org/wiki/Interval_%28mathematics%29#Notations_for_intervals

Reply:
You refer to the error message, right?

Reply:
I'm referring to "a float in the [0,1] range" in the error message